src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -74,6 +81,29 @@ class RsPathReferenceImpl(
     }
 }
 
+private fun filterPathCompletionVariantsByTraitBounds(
+    processor: RsResolveProcessor,
+    lookup: ImplLookup
+): RsResolveProcessor {
+    val cache = mutableMapOf<RsImplItem, Boolean>()
+    return fun(it: ScopeEntry): Boolean {
+        if (it !is AssocItemScopeEntry) return processor(it)
+        if (it.source !is TraitImplSource.ExplicitImpl) return processor(it)
+
+        val receiver = it.subst[TyTypeParameter.self()] ?: return processor(it)
+        // Don't filter partially unknown types
+        if (receiver.containsTyOfClass(TyUnknown::class.java)) return processor(it)
+
+        // Filter methods by trait bounds (try to select all obligations for each impl)
+        // We're caching evaluation results here because we can often complete to a methods
```

Comment:
Maybe `complete to a methods` -> `complete methods` (inside `filterMethodCompletionVariantsByTraitBounds` too)?